keps/sig-autoscaling/2702-graduate-hpa-api-to-GA/README.md

```diff
@@ -0,0 +1,511 @@
+`v1beta1.custom.metrics.k8s.io` and `v1beta1.external.metrics.k8s.io` APIs (usually served by the same adapter).
+Care should be taken to ensure the adapter and specific metrics backend is available to maintain custom metric autoscaling.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+All HPA objects are stored in v1 format on disk. They are up converted the requested version. And down converted upon update.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ x ] Events
+  - Event Reason: 
+  The event type `Normal`, reason `SuccessfulRescale`, note `New size: N; reason: FOO` indicates autoscaling is operating normally.
+  Abnormal events type `Warning` include reasons such as `FailedRescale` and `FailedComputeMetricsReplicas` and
+  will include details about the error in the note.
+- [ x ] API .status
+  - Condition name: 
+  There are three condition types which indicate the operating status of the HPA.  They are `ScalingEnabled`, `AbleToScale`
+  and `ScalingLimited` (see type [comments](https://pkg.go.dev/k8s.io/api/autoscaling/v2beta2#HorizontalPodAutoscalerConditionType))
+  Under normal operating circumstances `ScalingEnabled` and `AbleToScale` should be status `true`, indicating the HPA is
+  successfully reconciling the scale. `ScalingLimited` indicates user configuration is limiting the "ideal" scale with a
+  minimum, maximum, rate or delay. Which limit is the cause will be indicated in the message.
+  It's normal for this to be `true` or `false` periodically.
+  - Other field: 
+- [ x ] Other (treat as last resort)
+  - Details:
+  The HPA status includes the current observed metric values, one for each given target. Using these
+  values an operator can verify the HPA is maintaining the desired target for the dominant metric.
+  The operator can also see the number of pods the HPA observed under `status.currentReplicas` and the most
+  recent recommendation under `status.desiredReplicas`.
+  The latest observed generation is echoed back in status so an operator can verify the HPA is keeping up-to-date
+  with configuration changes.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
```

Comment:
Do you have SLOs/SLIs for HPA?

Reply:
No, we don't.

Reply:
So there are no metrics at all from the autoscaler? Probably something you want to add though I won't block this KEP for it.